desc/integrals/basis.py

```diff
@@ -58,6 +85,627 @@ def _in_epigraph_and(is_intersect, df_dy_sign, /):
     return put(is_intersect, idx[0], edge_case)
 
 
+def _chebcast(cheb, arr):
+    """Add leftmost axis to ``cheb`` depending on ``arr.ndim``.
+
+    Input ``arr`` should not have rightmost dimension of cheb that iterates
+    coefficients, but may have additional leftmost dimension for batch operation.
+    """
+    errorif(
+        jnp.ndim(arr) > cheb.ndim,
+        NotImplementedError,
+        msg=f"Only one additional axis for batch dimension is allowed. "
+        f"Got {jnp.ndim(arr) - cheb.ndim + 1} additional axes.",
+    )
+    return cheb if jnp.ndim(arr) < cheb.ndim else cheb[jnp.newaxis]
+
+
+class FourierChebyshevSeries(IOAble):
+    """Fourier-Chebyshev series.
+
+    f(x, y) = ∑ₘₙ aₘₙ ψₘ(x) Tₙ(y)
+    where ψₘ are trigonometric polynomials on [0, 2π]
+    and Tₙ are Chebyshev polynomials on [−yₘᵢₙ, yₘₐₓ].
+
+    Notes
+    -----
+    Performance may improve significantly
+    if the spectral resolutions ``M`` and ``N`` are powers of two.
+
+    Attributes
+    ----------
+    M : int
+        Fourier spectral resolution.
+    N : int
+        Chebyshev spectral resolution.
+    domain : (float, float)
+        Domain for y coordinates.
+    lobatto : bool
+        Whether ``f`` was sampled on the Gauss-Lobatto (extrema-plus-endpoint)
+        instead of the interior roots grid for Chebyshev points.
+
+    """
+
+    def __init__(self, f, domain=(-1, 1), lobatto=False):
+        """Interpolate Fourier-Chebyshev series to ``f``.
+
+        Parameters
```

Comment:
For all the other classes we don't document init, but make it part of the class docstring, I'd prefer that here as well